packages/mongodb-chatbot-server/src/routes/responses/createResponse.ts

```diff
@@ -506,22 +511,42 @@ const formatMessage = (
   metadata?: Record<string, string>
 ): MessagesParam[number] => {
   // store a placeholder string if we're not storing message data
-  const content = store ? message.content : "";
-  // handle cleaning custom data if we're not storing message data
-  const customData = {
-    ...message.customData,
-    query: store ? message.customData?.query : "",
-    reason: store ? message.customData?.reason : "",
-  };
+  const formattedContent = store ? message.content : "";
+  // handle cleaning metadata fields if we're not storing message data
+  const formattedMetadata = formatMetadata({ shouldStore: store, metadata });
+  const formattedCustomData = formatMetadata({
+    shouldStore: store,
+    metadata: message.customData,
+  });
 
   return {
     ...message,
-    content,
-    metadata,
-    customData,
+    content: formattedContent,
+    metadata: formattedMetadata,
+    customData: formattedCustomData,
   };
 };
 
+/** These metadata keys will persist in conversations and messages even if store is false.
+ * Otherwise, keys will have their values set to an empty string if store is false. */
+export const ALWAYS_ALLOWED_METADATA_KEYS = ["ip", "origin", "userAgent"];
+
+interface FormatMetadataParams {
```

Comment:
this isn't passed to the constructor function, is it?

Reply:
done!